contracts/registry/implementation/IdentityRegistry.sol

```diff
@@ -268,8 +268,11 @@ contract IdentityRegistry is IIdentityRegistry, AgentRoleUpgradeable, IRStorage
         IIdentity _identity,
         uint16 _country
     ) public override onlyAgent {
-        _tokenIdentityStorage.addIdentityToStorage(_userAddress, _identity, _country);
-        emit IdentityRegistered(_userAddress, _identity);
+        IIdentity identity = _tokenIdentityStorage.storedIdentity(_userAddress);
```

Comment:
you shouldn't change the registerIdentity method for updating the recovery process

Reply:
if a wallet is already registered there is a function update to change the OID linked to it, add function shouldn't be used for that